src/game.cpp

```diff
@@ -1791,8 +1791,110 @@ void game::update_weather()
 
 int game::get_temperature( const tripoint &location )
 {
+    int temp_mod = 0; // local modifier
+    
+    if( !new_game ) {
+        // Direct heat from fire sources
+        // Cache fires to avoid scanning the map around us bp times
+        // Stored as intensity-distance pairs
+        std::vector<std::pair<int, int>> fires;
+        fires.reserve( 13 * 13 );
+        int best_fire = 0;
+        for( const tripoint &dest : g->m.points_in_radius( location, 6 ) ) {
+            int heat_intensity = 0;
+
+            int ffire = g->m.get_field_strength( dest, fd_fire );
+            if( ffire > 0 ) {
+                heat_intensity = ffire;
+            } else if( g->m.tr_at( dest ).loadid == tr_lava ) {
+                heat_intensity = 3;
+            }
+            if( heat_intensity == 0 || !g->m.sees( location, dest, -1 ) ) {
+                // No heat source here
+                continue;
+            }
+            // Ensure fire_dist >= 1 to avoid divide-by-zero errors.
+            const int fire_dist = std::max( 1, square_dist( dest, location ) );
+            fires.emplace_back( std::make_pair( heat_intensity, fire_dist ) );
+            if( fire_dist <= 1 ) {
+                // Extend limbs/lean over a single adjacent fire to warm up
+                best_fire = std::max( best_fire, heat_intensity );
+            }
+        }
+
+        for( const auto &intensity_dist : fires ) {
+            const int intensity = intensity_dist.first;
+            const int distance = intensity_dist.second;
+            temp_mod = 6 * intensity * intensity / distance;
```

Comment:
value of `temp_mod` on exiting this loop is always the value of `6 * fires.back().first * fires.back().first / fires.back().second` and effectively ignores all other values. Is this intentional?

Reply:
You are right, should be `+=`. Heat radiation should sum up from multiple sources, thats the sole purpose of this loop.

Reply:
Is it supposed to be `6* (intensity^2 / distance)`? I think radiant heat would follow an inverse-square intensity change more closely ( `6 * (intensity / distance^2)` ), dropping off rapidly with distance. A linear drop isn't an issue though, mostly concerned about the intensity being squared.